components/automate-ui/src/app/pages/+compliance/shared/reporting/report-query.service.ts

```diff
@@ -89,4 +119,33 @@ export class ReportQueryService {
   getFilterTitle(type: string, id: string): string {
     return this.idToTitle.get(type + '-' + id);
   }
+
+  formatReturnParams(): ReturnParams {
+    const reportQuery = this.getReportQuery();
+
+    const structuredFilters = {};
+    reportQuery.filters.map(filter => {
+      return structuredFilters[filter.type.name] = filter.value.text;
+    });
+
+    const today = moment().utc()
+                      .startOf('day').add(12, 'hours'); // must sync with initial end Date
+    const endDate = reportQuery.endDate
+      ? moment(reportQuery.endDate).format(DateTime.REPORT_DATE)
+      : today.format(DateTime.REPORT_DATE);
+
+    // Check if today is the end date
+    const isToday = today.diff(endDate, 'days') === 0;
+
+    // With no filters when the date it today, we don't want
+    // to add any parameters to the URL.
+    if (isEmpty(structuredFilters) && isToday) {
+      return {};
+    } else {
+      return {
+        end_time: endDate,
+        ...structuredFilters
+      };
```

Comment:
I cannot infer anything about possible values for `structuredFilters` so better to be safe by spreading first, to guarantee our `endDate` value is used.
```suggestion
      return { ...structuredFilters, end_time: endDate };
```
(If `structuredFilters` is really `ReportingFilterTypes`, then definitely need the change.)

Reply:
thanks!  I updated this now and added stronger typing to StructuredFilters